cephfs/cephfs_test.go

```diff
@@ -15,14 +16,60 @@ import (
 )
 
 var (
-	CephMountTest = "/tmp/ceph/mds/mnt/"
+	CephMountDir     = "/tmp/ceph/mds/mnt/"
+	requireCephMount = false
+	testMdsName      = "Z"
 )
 
+func init() {
+	mdir := os.Getenv("GO_CEPH_TEST_MOUNT_DIR")
+	if mdir != "" {
+		CephMountDir = mdir
+	}
+	reqMount := os.Getenv("GO_CEPH_TEST_REQUIRE_MOUNT")
+	if reqMount == "yes" || reqMount == "true" {
+		requireCephMount = true
+	}
+	mdsName := os.Getenv("GO_CEPH_TEST_MDS_NAME")
+	if mdsName != "" {
+		testMdsName = mdsName
+	}
+}
+
+func useMount(t *testing.T) {
+	fail := func(m string) {
+		if requireCephMount {
+			t.Fatalf("cephfs mount required: %s %s", CephMountDir, m)
+		} else {
+			t.Skipf("cephfs mount needed: %s %s", CephMountDir, m)
+		}
+	}
+
+	s, err := os.Stat(CephMountDir)
+	if err != nil || !s.IsDir() {
+		fail("missing or not a directory")
+	}
+
+	if us, ok := s.Sys().(*syscall.Stat_t); ok {
```

Comment:
Oh dear, that looks ugly. But it has to be written like that I suppose? Actually I don't understand the syntax. And what does `us` stand for ?

Reply:
Yeah, blame the Go os package for that. They seem to be rather overzealous about hiding differences between platforms in the filesystems related code. So to do what I consider normal posixy stuff you've got to do contortions.

The s is the "platform agnostic stat" object. Calling the `Sys()` function returns a platform specific object as an `interface{}` so we have to type convert it into the right type for a unixy platform which is `*syscall.Stat_t`. "us" just stands for "unix stat".

Reply:
Oh, it's a type _assertion_, I somehow didn't recognize the syntax here. Got it now. Still ugly. ;-)